### PR TITLE
testing/ostest: Fix the compiler warning

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
## Summary
```
aio.c: In function ‘check_done’:
aio.c:158:41: warning: format ‘%d’ expects argument of type ‘int’, but argument 3 has type ‘ssize_t’ {aka ‘volatile long int’} [-Wformat=]
  158 |           printf("  list[%d]. result = %d\n", i, aiocbp->aio_result);
      |                                        ~^        ~~~~~~~~~~~~~~~~~~
      |                                         |              |
      |                                         int            ssize_t {aka volatile long int}
      |                                        %ld
aio.c: In function ‘remove_done’:
aio.c:222:41: warning: format ‘%d’ expects argument of type ‘int’, but argument 3 has type ‘ssize_t’ {aka ‘volatile long int’} [-Wformat=]
  222 |           printf("  list[%d]. result = %d\n", i, aiocbp->aio_result);
      |                                        ~^        ~~~~~~~~~~~~~~~~~~
      |                                         |              |
      |                                         int            ssize_t {aka volatile long int}
      |                                        %ld

sporadic2.c: In function ‘sporadic2_test’:
sporadic2.c:313:10: warning: format ‘%i’ expects argument of type ‘int’, but argument 4 has type ‘long int’ [-Wformat=]
  313 |   printf("Sporadic 1: prio high %d, low %d, repl %" PRIi32 " ns\n",
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /home/xiaoxiang/backup/os/nuttx/nuttx/include/stdint.h:35,
                 from /home/xiaoxiang/backup/os/nuttx/nuttx/include/sys/types.h:32,
                 from /home/xiaoxiang/backup/os/nuttx/nuttx/include/stdio.h:30,
                 from sporadic2.c:27:
/home/xiaoxiang/backup/os/nuttx/nuttx/include/arch/inttypes.h:51:24: note: format string is defined here
   51 | #  define PRIi32      "i"
sporadic2.c:315:10: warning: format ‘%i’ expects argument of type ‘int’, but argument 4 has type ‘long int’ [-Wformat=]
  315 |   printf("Sporadic 2: prio high %d, low %d, repl %" PRIi32 " ns\n",
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

## Impact
Minor

## Testing
sim:ostest

### CODE REVIEW

File: testing/ostest/aio.c [hunk 1]
```diff
@@ -155,7 +155,7 @@ static int check_done(void)
         {
           /* Check if the I/O has completed */
 
-          printf("  list[%d]. result = %d\n", i, aiocbp->aio_result);
+          printf("  list[%d]. result = %ld\n", i, aiocbp->aio_result);
```
Comment (gustavonihei) [inline testing/ostest/aio.c:158]:
```suggestion
          printf("  list[%d]. result = %zd\n", i, aiocbp->aio_result);
```
`aio_result` type is `ssize_t`, so I think `z` fits best here.

File: testing/ostest/aio.c [hunk 2]
```diff
@@ -219,7 +219,7 @@ static int remove_done(void)
...
```
Comment (gustavonihei) [inline testing/ostest/aio.c:222]:
```suggestion
          printf("  list[%d]. result = %zd\n", i, aiocbp->aio_result);
```
`aio_result` type is `ssize_t`, so I think `z` fits best here.

File: testing/ostest/sporadic2.c [hunk 3]
```diff
@@ -310,9 +310,9 @@ int sporadic2_test(void)
   int32_t budget_2_ns = 30 * 1000000;
   int i;
 
-  printf("Sporadic 1: prio high %d, low %d, repl %" PRIi32 " ns\n",
+  printf("Sporadic 1: prio high %d, low %d, repl %ld ns\n",
```
Comment (gustavonihei) [inline testing/ostest/sporadic2.c:313]: How about changing the definition of REPL_INTERVAL to:
```patch
--- a/testing/ostest/sporadic2.c
+++ b/testing/ostest/sporadic2.c
@@ -44,7 +44,7 @@
 #define PRIO_MID      100
 #define PRIO_HIGH1    180
 #define PRIO_HIGH2    170
-#define REPL_INTERVAL 100000000L
+#define REPL_INTERVAL INT32_C(100000000)
 #define MAX_BUDGET    (REPL_INTERVAL / 2)
 #define PRIO_HI_NDX   0
 #define PRIO_LO_NDX   1
 ```
Then we may keep the format specifier as it currently is.

File: testing/ostest/aio.c (hunk 2 above)
Comment (xiaoxiang781216) [reply testing/ostest/aio.c]: Done.

File: testing/ostest/aio.c (hunk 1 above)
Comment (xiaoxiang781216) [reply testing/ostest/aio.c]: Done.

File: testing/ostest/sporadic2.c (hunk 3 above)
Comment (xiaoxiang781216) [reply testing/ostest/sporadic2.c]: Done.

**Comment by xiaoxiang781216** [comment]
@gustavonihei CI break is fixed by: https://github.com/apache/incubator-nuttx/pull/4751

**Review by gustavonihei — APPROVED** [review]
LGTM

**Pull request merged**